pkg/skaffold/util/util.go

```diff
@@ -287,6 +287,14 @@ func IsDir(path string) bool {
 	return (err == nil || !os.IsNotExist(err)) && info.IsDir()
 }
 
+func IsEmptyDir(path string) bool {
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return false
+	}
+	return len(files) == 0
```

Comment:
I think it would be worth optimizing this.  The [implementation of `os.ReadDir(string)`](https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/os/dir.go;l=115-125) does the following:
```golang
func ReadDir(name string) ([]DirEntry, error) {
	f, err := Open(name)
	if err != nil {
		return nil, err
	}
	defer f.Close()

	dirs, err := f.ReadDir(-1)
	sort.Slice(dirs, func(i, j int) bool { return dirs[i].Name() < dirs[j].Name() })
	return dirs, err
}
```

We don't care about the full list nor about sorting.  So we could just use:
```suggestion
	d, err := os.Open(name)
	if err != nil {
		return false
	}
	defer d.Close()
	ents, err := d.ReadDir(1)
	if err != nil {
		return false
	}
	return len(ents) == 0
```

Reply:
Agree. That will be more efficient for this use case.